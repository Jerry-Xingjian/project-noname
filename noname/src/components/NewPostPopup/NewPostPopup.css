.modal {
  position: relative;
  z-index: 1;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.25);
}

.modal_content {
  background-color: white;
  position: absolute;
  top: 50%;
	left: -12.5%;
  width: 125%;
  padding: 20px;
  border-radius: 5px;
  border: 2px solid black;
}

#dismiss {
	position: absolute;
	left:44%;
}

#dismiss:hover {
  cursor:pointer;
}


.comment-input {
	width: 75%;
}

.location {
  width: 50%;
}

#post-button {
	position: relative;
	width: 120px;
	border-radius: 4px;
	color: #cf250D;
	border: solid 2px transparent;
	background: linear-gradient(rgba(255, 255, 255, 0), rgba(255, 255, 255, 0)), linear-gradient(61deg, #cf250D, #ffae00);
	background-origin: border-box;
	background-clip: content-box, border-box;
	box-shadow: 2px 1000px 1px #fff inset;

}

#images-input {
	display: none;
}

#images-input-box {
	position: relative;
	width: 75%;
	left: -9.5%;
	border-radius: 5px;
	border-color: red;
}
