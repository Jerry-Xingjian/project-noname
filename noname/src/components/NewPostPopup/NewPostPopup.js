import React from 'react';
// import AWS from 'aws-sdk';
import Modal from 'react-bootstrap/Modal';
import ImageUploader from '../ImageUploader/ImageUploader';
import './NewPostPopup.css';
import { addPost } from '../../utils/api/post';
import uploadFile from '../../utils/api/aws';

function NewPostPopup(props) {
  const {
    show,
    onHide,
    userId,
    fetchPosts,
  } = props;

  let media = [];

  const onImagesUpload = (images) => {
    media = [];
    images.map((image) => (
      media.push(uploadFile(image.file))
    ));
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    if (media.length === 0) {
      // alert('No file uploaded!');
      document.getElementById('alert-box').innerHTML = 'Please upload a photo!';
      return;
    }
    const location = document.getElementById('location-input-newpost').value;
    const caption = document.getElementById('comment').value;
    await addPost(userId, media, caption, location).then(fetchPosts());
    onHide();
    window.location.reload(false);
  };

  return (
    <Modal
      show={show}
      onHide={onHide}
      className="modal"
    >
      <div className="modal_content">
        <div className="container my-3 " data-testid="new-post-popup">
          <div className="row">
            <p className="text-left my-1">Create New Post</p>
            <svg className="my-1" id="dismiss" onClick={onHide} width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
              <path d="M9.16992 14.8299L14.8299 9.16992" stroke="#292D32" strokeWidth="1.5" strokeLinecap="round" strokeLinejoin="round" />
              <path d="M14.8299 14.8299L9.16992 9.16992" stroke="#292D32" strokeWidth="1.5" strokeLinecap="round" strokeLinejoin="round" />
              <path d="M9 22H15C20 22 22 20 22 15V9C22 4 20 2 15 2H9C4 2 2 4 2 9V15C2 20 4 22 9 22Z" stroke="#292D32" strokeWidth="1.5" strokeLinecap="round" strokeLinejoin="round" />
            </svg>
            <hr className="divider" />
            <div className="new-post-input mx-5 my-3">
              <form onSubmit={handleSubmit}>
                <div className="container border py-2" id="images-input-box">
                  <ImageUploader onImagesUpload={onImagesUpload} />
                  <div className="container" id="alert-box" />
                </div>
                <textarea data-testid="new-post-caption" name="comment" id="comment" rows="5" placeholder="Add captions here..." className="comment-input mx-3 my-4" />
                <br />
                <input
                  type="text"
                  placeholder="Location"
                  name="location-input-newpost"
                  id="location-input-newpost"
                  className="location mx-3 my-2"
                />
                <input data-testid="new-post-button" type="submit" value="POST" className="post" id="post-button" />
              </form>
            </div>
          </div>
        </div>
      </div>
    </Modal>
  );
}

export default NewPostPopup;
