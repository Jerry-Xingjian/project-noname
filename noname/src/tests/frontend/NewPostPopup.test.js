import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import '@testing-library/jest-dom'
import NewPostPopup from "../../components/NewPostPopup/NewPostPopup";
import renderer from 'react-test-renderer';

describe("Snapshot test", () => {
	test("Post card structure test", async () => {
		const component = renderer.create(
			<NewPostPopup />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});


describe("element test", () => {
	test("test UI elements ", async () => {
		// test for different effects than the page actually changing
		render(
			<NewPostPopup show={true}/>
		)
		await waitFor(() => {
			expect(screen.getByText(/Upload/i)).toBeInTheDocument()
		})
	});
});

describe("User input event test", () => {
	test("textbox have text after user input", async () => {
		render(<NewPostPopup show={true} />);
		const element = screen.getByTestId('new-post-caption');
		fireEvent.change(element, { target: { value: 'hello'}});
		expect(element.value).toBe('hello');
	})
});

describe("Submit button event test", () => {
	const defaultProp = {
		show: true,
		onHide: ()=>console.log('hide'),
		userId: "111",
		fetchPosts: jest.fn()
	}
	test('click on post button', async () => {
		render(<NewPostPopup  {...defaultProp}/>);
		let text = screen.getByTestId('new-post-caption');
		expect(text).toHaveValue('');
		fireEvent.change(text, {target: {value: 'I love this post'}});
		expect(text).toHaveValue('I love this post');
		const element = screen.getByTestId('new-post-button');
		fireEvent.click(element);
		// expect(text).toHaveValue('');
	})
});

