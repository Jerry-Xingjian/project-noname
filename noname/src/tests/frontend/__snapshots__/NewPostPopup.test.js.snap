// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Post card structure test 1`] = `null`;
